packages/cardpay-sdk/sdk/utils/general-utils.ts

```diff
@@ -327,18 +327,24 @@ export function isTransactionHash(candidate: string): boolean {
 }
 
 export async function sendTransaction(web3: Web3, transaction: Transaction): Promise<string> {
-  /* eslint-disable no-async-promise-executor */
-  let txHash: string = await new Promise(async (resolve, reject) => {
+  let from = (await web3.eth.getAccounts())[0];
+  return await new Promise((resolve, reject) => {
     web3.eth
-      .sendTransaction({
-        ...transaction,
-        from: (await web3.eth.getAccounts())[0],
-      })
-      .once('transactionHash', (transactionHash) => resolve(transactionHash))
-      .catch((err) => reject(err));
+      .sendTransaction(
+        {
+          ...transaction,
+          from,
+        },
+        (err, hash) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(hash);
+          }
+        }
+      )
+      .catch(reject);
```

Comment:
The sematics of this are different. now. previously this would return as soon as a txn hash was available, now i think it will wait for a consensus before it returns. I don't think we want that. rather we should only wait until a txnHash is available. (when the `transactionHash` event is emitted). check out the docs for more info: https://web3js.readthedocs.io/en/v1.8.0/web3-eth.html#eth-sendtransaction-return

Reply:
I reverted these changes because we migrated from `web3` to `ethers.js` for the scheduled payment module.